File: iteration2/Controllers/Controller.java
package iteration2.Controllers;

import iteration2.Models.*;
import org.json.JSONObject;
import org.json.simple.parser.JSONParser;

import java.io.*;
import java.util.ArrayList;

public abstract class Controller {
    private RegistrationError error;

    public ArrayList<JSONObject> readJSONFiles(String path) {
        ArrayList<JSONObject> jsonObjects = new ArrayList<>();

        String requestedPath = System.getProperty("user.dir") + "/iteration2/Data/Input/" + path;
        String[] fileNames = new File(requestedPath).list();

        for (String file : fileNames) {
            try {
                Object obj = new JSONParser().parse(new FileReader(requestedPath + "/" + file));
                JSONObject jo = new JSONObject(obj.toString());
                jsonObjects.add(jo);
            } catch (Exception e) {
                System.out.println(e.getMessage());
            }
        }

        return jsonObjects;
    }

    public boolean exportJSONFile(Object object) {
        String fileName = "";
        String path = "";
        JSONObject jsonObject = null;
        String content = "";

        switch (object.getClass().getSimpleName()) {
            case "Student":
                fileName = ((Student) object).getId();
                jsonObject = ((Student) object).toJson();
                path = "Output/Students";
                content = jsonObject.toString();
                break;
            case "Advisor":
                fileName = (((Advisor) object).getName() + ((Advisor) object).getSurname()).replace(" ", "");
                jsonObject = ((Advisor) object).toJson();
                path = "Output/Advisors";
                content = jsonObject.toString();
                break;
            case "RegistrationError":
                fileName = "RegistrationErrors";
                jsonObject = ((RegistrationError) object).toJson();
                content = jsonObject.toString();
                break;
            default:
                return false;
        }
        String fullFileName = System.getProperty("user.dir") + "/iteration2/Data/" + path + "/" + fileName + ".json";

        try {
            File directory = new File(String.valueOf(System.getProperty("user.dir") + "/iteration2/Data/" + path));

            if (!directory.exists()) {
                directory.mkdirs();
            }
            //System.exit(1);
            if (!directory.exists()) {
                directory.mkdir();
            }
            File myObj = new File(fullFileName);
            myObj.createNewFile();

            FileWriter myWriter = new FileWriter(fullFileName);
            myWriter.write(content);
            myWriter.close();
        } catch (IOException e) {
            System.out.println("An error occurred while exporting .json file.");
            e.printStackTrace();
        }
        return false;
    }

    public RegistrationError getError() {
        return error;
    }

    public void setError(RegistrationError error) {
        this.error = error;
    }
}

File: iteration3/Controllers/Controller.py
import json
import os

from iteration3.Models.RegistrationError import RegistrationError


class Controller:
    def __init__(self):
        self.__error = None

    def read_json_files(self, path):
        json_objects = []
        requested_path = os.getcwd() + '/Data/Input/' + path
        print("--C--->Reading json files on path: " + requested_path)
        file_names = os.listdir(requested_path)

        for file in file_names:
            try:
                with open(requested_path + '/' + file) as f:
                    obj = json.load(f)
                    jo = json.loads(json.dumps(obj))
                    json_objects.append(jo)
            except Exception as e:
                print(e)

        return json_objects

    def export_json_file(self, object):
        path = ""

        if object.__class__.__name__ == "Student":
            file_name = object.id
            json_object = object.to_json()
            path = "Output/Students"
            content = json_object
        elif object.__class__.__name__ == "Advisor":
            file_name = object.name + object.surname
            file_name = file_name.replace(" ", "")
            json_object = object.to_json()
            path = "Output/Advisors"
            content = json_object
        elif object.__class__.__name__ == "RegistrationError":
            file_name = "RegistrationErrors"
            json_object = object.to_json()
            content = json_object
        else:
            return False

        full_file_name = os.getcwd() + '/Data/' + path + '/' + str(file_name) + '.json'

        try:
            directory = os.path.dirname(full_file_name)

            if not os.path.exists(directory):
                os.makedirs(directory)

            with open(full_file_name, 'w') as f:
                json.dump(content, f)
                # f.write(content)
        except OSError as e:
            print("An error occurred while exporting .json file.")
            print(e)
        return False

    @property
    def error(self):
        return self.__error

    @error.setter
    def error(self, error):
        self.__error = error
